test/test_nn.py

```diff
@@ -52,6 +54,8 @@
               torch.double: 1e-5,
               torch.half: 1e-2}
 
+NUM_SHARDS = 30
```

Comment:
It would be better to avoid hardcoding this one in the script and make it an argument too. This might be useful even for local development

Reply:
I agree. We can have 30 as default value.